processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java

```diff
@@ -609,6 +609,14 @@ public static Dictionary getDictionary(AbsoluteTableIdentifier absoluteTableIden
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_SIZE_FIRST;
         } else {
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_NUM_FIRST;
+          // fall back to BLOCK_NUM_FIRST strategy need to reset
+          // the average expected size for each node
+          if (blockInfos.size() > 0) {
```

Comment:
```suggestion
          if (numOfNodes > 0) {
```
if  blockInfos.size() = 0 ， sizePerNode will be 0， so no need to add if ... else ... 
Do numOfNodes need to consider be 0?

Reply:
could be set to some value if use NODE_MIN_SIZE_FIRST but fall back to BLOCK_NUM_FIRST